### ISSUE TITLE
Embeddings column type and vector similarity search

### ISSUE THREAD

**Issue by nileshtrivedi** [teableio/teable#1163]
I would like to be able to perform similarity search over vector embeddings generated by language models to find records in a table.

**Describe the solution you'd like**
Postgresql has an extension [pgvector](https://github.com/pgvector/pgvector) which allows easy storage and query over embedding vectors.

It supports:
    - exact and approximate nearest neighbor search
    - single-precision, half-precision, binary, and sparse vectors
    - L2 distance, inner product, cosine distance, L1 distance, Hamming distance, and Jaccard distance
    - any [language](https://github.com/pgvector/pgvector#languages) with a Postgres client

Sample usage in SQL:
```
CREATE EXTENSION vector;
-- Create a vector column with 3 dimensions
CREATE TABLE items (id bigserial PRIMARY KEY, embedding vector(3));

-- Insert Vectors
INSERT INTO items (embedding) VALUES ('[1,2,3]'), ('[4,5,6]');

-- Get the nearest neighbors by L2 distance
SELECT * FROM items ORDER BY embedding <-> '[3,1,2]' LIMIT 5;
```

pgvector supports inner product (<#>), cosine distance (<=>), and L1 distance (<+>, added in 0.7.0)

These operations are agnostic of the model which was used to generate these embedding vectors.

**Comment by ikx94** [comment]
This would be great

**Comment by nileshtrivedi** [comment]
@caoxing9 I would like to take a stab at this. Prisma docs [give an example](https://www.prisma.io/docs/postgres/database/postgres-extensions#overview) of how to add vector type in postgres (via pgvector extension). Though, I am not sure how vector column types will work for sqlite or Turso/libSQL.

I am not familiar with Teable internals. How should I go about implementing this?

**Comment by nileshtrivedi** [comment]
> Native pgvector support in Prisma ORM is coming soon. In the meantime, use it via [custom migrations](https://www.prisma.io/docs/orm/prisma-migrate/workflows/customizing-migrations) and [TypedSQL](https://www.prisma.io/blog/announcing-typedsql-make-your-raw-sql-queries-type-safe-with-prisma-orm).

From https://www.prisma.io/blog/orm-6-13-0-ci-cd-workflows-and-pgvector-for-prisma-postgres

**Comment by caoxing9** [comment]
Hi @nileshtrivedi , thanks for your interest in contributing! 🙌

Before diving in, could you share more about your use case? Are you looking to build something like AI-powered semantic search?

Regarding SQLite / Turso / libSQL — no worries there! SQLite is only used for local development and is completely optional, so you can safely skip it and focus on the PostgreSQL (pgvector) implementation.

Looking forward to hearing more about what you have in mind!

**Comment by nileshtrivedi** [comment]
@caoxing9 Yes, I like to use self-hosted Teable against my private databases. Semantic search over documents or other objects is the immediate use-case. At work, we use Postgresql for customers' apps. While we are not using Teable at work yet, having this ( and #876 ) will help me make a case for it internally.

**Comment by caoxing9** [comment]
> [@caoxing9](https://github.com/caoxing9) Yes, I like to use self-hosted Teable against my private databases. Semantic search over documents or other objects is the immediate use-case. At work, we use Postgresql for customers' apps. While we are not using Teable at work yet, having this ( and [#876](https://github.com/teableio/teable/issues/876) ) will help me make a case for it internally.

Thanks for the feedback and sharing your use case! These features aren't on our immediate roadmap at the moment, but we'll revisit and discuss them once we've wrapped up our current priorities
